synapse/tools/migrate_010.py

```diff
@@ -0,0 +1,715 @@
+import time
+import logging
+import pathlib
+import argparse
+import tempfile
+from binascii import unhexlify, hexlify
+
+import lmdb  # type: ignore
+
+import synapse.cortex as s_cortex
+import synapse.lib.lmdb as s_lmdb
+import synapse.lib.msgpack as s_msgpack
+import synapse.models.inet as s_inet
+
+logger = logging.getLogger(__name__)
+
+# Topologically sorted comp and sepr types that are form types that have other comp types as elements.  The beginning
+# of the list has more dependencies than the end.
+_comp_and_sepr_forms = [
+    'tel:mob:imsiphone', 'tel:mob:imid', 'syn:tagform', 'syn:fifo', 'syn:auth:userrole', 'seen', 'rsa:key', 'recref',
+    'ps:image', 'ou:user', 'ou:suborg', 'ou:member', 'ou:meet:attendee', 'ou:hasalias', 'ou:conference:attendee',
+    'mat:specimage', 'mat:itemimage', 'it:hostsoft', 'it:dev:regval', 'it:av:filehit', 'it:av:sig',
+    'inet:wifi:ap', 'inet:whois:regmail', 'inet:whois:recns', 'inet:whois:contact', 'inet:whois:rec',
+    'inet:web:post', 'inet:web:mesg', 'inet:web:memb', 'inet:web:group', 'inet:web:follows', 'inet:web:file',
+    'inet:web:acct', 'inet:urlredir', 'inet:urlfile', 'inet:ssl:tcp4cert', 'inet:servfile', 'inet:http:resphead',
+    'inet:http:reqparam', 'inet:http:reqhead', 'inet:http:param', 'inet:http:header', 'inet:dns:txt',
+    'inet:dns:soa', 'inet:dns:rev6', 'inet:dns:rev', 'inet:dns:req', 'inet:dns:ns', 'inet:dns:mx',
+    'inet:dns:cname', 'inet:dns:aaaa', 'inet:dns:a', 'inet:asnet4', 'geo:nloc', 'file:subfile']
+
+def _enc_iden(iden):
+    return unhexlify(iden)
+
+class ConsistencyError(Exception):
+    pass
+
+def _special_sort_key(t):
+    '''
+    For processing secondary properties in order, always processing tcp after udp, and v4 after v6
+    '''
+    k, v = t
+    k = k.replace('tcp', chr(127) + 'tcp', 1)
+    return (k.replace('v4', chr(127) + 'v4', 1), v)
+
+class Migrator:
+    '''
+    Sucks all rows out of a < .0.1.0 cortex, into a temporary LMDB database, migrates the schema, then dumps to new
+    file suitable for ingesting into a >= .0.1.0 cortex.
+    '''
+    def __init__(self, core, outfh, tmpdir=None, stage1_fn=None, rejects_fh=None):
+        '''
+        Create a migrator.
+
+        Args:
+            core (synapse.cores.common.Cortex): 0.0.55 cortex to export from
```

Comment:
`0.0.x cortex`

Reply:
need to specify that this must be a local cortex and not a telepath proxy

Reply:
ok